WATOLINK/bci-gaming
# Player ranged attack

## ClickUp ticket link

[Player Long Range Attack](https://app.clickup.com/t/8685qfv3d)



## Implementation description
Prefab shooting is implemented for player ranged attacks.
Bullet implementation could be found in TemplateProjectile, and impact effect implementation could be found in Temporary Projectile Effect.
The bullet shoots from the direction where the player is facing by using the RangeFirePoint.



## Steps to test
1. Play game and press down on Ctrl button
2. Make sure that the bullet is shot after the player animation is finished, and that the bullet hits the target (either enemy or platform). If there is no obstacle in the direction of the player, then the bullet should "fly" offscreen
3. Make sure that the instantiated bullet and impact effect (the effect when a bullet hits a target) prefabs would be deleted from the hierarchy once the ranged attack is finished.
4. If enemy is hit, ensure that enemy health decreases



## What should reviewers focus on?
Ranged attack mechanics should run smoothly


## Checklist
